src/fragmentarium/ui/SearchGroup.tsx

```diff
@@ -1,28 +1,74 @@
-import React, { FunctionComponent } from 'react'
+import React, { Component } from 'react'
 import NumberSearchForm from 'fragmentarium/ui/search/NumberSearchForm'
 import TransliterationSearchForm from 'fragmentarium/ui/search/TransliterationSearchForm'
 import LuckyButton from 'fragmentarium/ui/front-page/LuckyButton'
 import PioneersButton from 'fragmentarium/ui/PioneersButton'
-
+import { withRouter, RouteComponentProps } from 'react-router-dom'
 import './SearchGroup.css'
+import { Button } from 'react-bootstrap'
+import { stringify } from 'query-string'
+import ReferenceSearchForm from './search/ReferenceSearchForm'
+import {
+  FragmentariumSearchParams,
+  SearchGroupParams,
+} from '../domain/fragmentariumSearch'
+
+type State = SearchGroupParams
+
+type Props = FragmentariumSearchParams & RouteComponentProps
+
+class SearchGroup extends Component<Props, State> {
+  constructor(props) {
+    super(props)
+    this.state = {
+      number: this.props.number || '',
+      title: this.props.title || '',
+      id: this.props.id || '',
+      pages: this.props.pages || '',
+      transliteration: this.props.transliteration || '',
+    }
```

Comment:
[Avoid copying props into state.](https://reactjs.org/docs/react-component.html#constructor)

Reply:
But we need some kind of State for storing the props so that we can manipulate them from the children classes (we cant manipulate the props directly)

Reply:
This was handled previously handled via the query string. You can leave the state for now if it does not cause any issues.